File: python/matrixmath.py
#Scalar Addition
def add (A, p):
#    result = {}{}
    if checkMat(A) == True:
        for i in A:
            for j in A[i]:
                result[i][j] = A[i][j] + p
    else:
        print "The input A is an invalid matrix"
    return result

#Scalar Addition
def add (p, A):
#    result = {}{}
    if checkMat(A) == True:
        for i in A:
            for j in A[i]:
                result[i][j] = A[i][j] + p
    else:
        print "The input A is an invalid matrix"
    return result

#Scalar Multiplication
def mult (A, p):
#    result = {}{}
    if checkMat(A) == True:
        for i in A:
            for j in A[i]:
                result[i][j] = A[i][j] * p
    else:
        print "The input A is an invalid matrix"
    return result

#Scalar Multiplication
def mu3lt (p, A):
#    result = {}{}
    if checkMat(A) == True:
        for i in A:
            for j in A[i]:
                result[i][j] = A[i][j] * p
    else:
        print "The input A is an invalid matrix"
    return result

#Matrix addition
def add (A, B):
#    result = {}{}
    if sameSize(A, B) == True:
        for i in A:
            for j in A[i]:
                result[i][j] = A[i][j] + B[i][j]
    else:
        print "A and B cannot be added together"
    return result

#Matrix Multiplication
def mult (A, B):
#    result = {}{}
    if checkMat(A) == True and checkMat(B) == True and canMult(A, B):
        multSum = 0
        for i in A:
            for j in A[i]:
                for k in A[i]:
                    multSum += (A[i][j] * B[i][j])
                result[i][j] = multSum
                multSum = 0
    else:
        print "A and B cannot be multiplied"
    return result

#Checks to make sure A is a valid matrix
def checkMat (A):
#    bool result = False
    i = 0
    while i < len(A):
        if len(A[i]) == len(A[0]):
            result = True
        else:
            result = False
            break
        i += 1
    if result == True:
        print "Matrix is valid"
    else:
        print "Matrix is invalid"

    return result

#Checks to see if two matrices are the same dimensions
def sameSize (A, B):
#    bool result = False
    if checkMat(A) == True and checkMat(B) == True:
        if len(A) == len(B) and len(A[0]) == len(B[0]):
            result = True
    else:
        result = False
    return result

#Checks to see if two matrices can be multiplied
def canMult (A, B):
#    bool result = False
    if checkMat(A) == True and checkMat(B) == True and len(A[0]) == len(B):
        result = True
    else:
        result = False
    return result

### Main Function ###

print "hello world"
A = [[1,3,5],
     [3,8,3],
     [1,9,3]]
p = 2
resultArray = add (A, p)

for i in A:
    for j in A[i]:
        print ("\t" + resultArray[i][j])
    print "\n"

File: newWorkspace/apcsProject/src/matrixmath.java
/*
Cody Brooks
2014-11-12
*/

public class matrixmath {

  public static void main (String[] args){
	  /*Setup inputs*/
	  int[][] A = {
			  {2,5,7},
			  {7,3,9},
			  {7,2,8}
	  };
	  int[][] B = {
			  {3,6,9},
			  {2,2,8},
			  {1,6,7}
	  };
	  int p = 7;

	  int[][] resultArray = MatMath.mult(A, p);

	  /*Print the result array*/
	  System.out.println("Result Array:");
	  for (int i = 0; i < resultArray.length; i++) {
		  for (int j = 0; j < resultArray[0].length; j++) {
			  System.out.print("\t" + resultArray[i][j]);
		  }
		  System.out.println();
	  }

  }

}

class MatMath {

	/*Scalar Addition*/
	static int [][] add (int p, int[][] A){
	    int[][] result = new int[A.length][A[0].length];
		if (checkMat(A)){
		    for (int i = 0; i < A.length; i++) {
		      for (int j = 0; j < A[i].length; j++) {
		        result[i][j] = p + A[i][j];
		      }
		    }
		} else {
			System.out.println("The input A is not a valid matrix");
			//Throw an exception
		}

    return result;
	}

  /*Scalar Addition*/
  static int [][] add (int[][] A, int p){
    int[][] result = new int[A.length][A[0].length];
	if (checkMat(A)){
	    for (int i = 0; i < A.length; i++) {
	      for (int j = 0; j < A[i].length; j++) {
	        result[i][j] = p + A[i][j];
	      }
	    }
	} else {
		System.out.println("The input A is not a valid matrix");
		//Throw an exception
	}

	return result;
  }

  /*Scalar Multiplication*/
  static int [][] mult (int p, int[][] A){
    int[][] result = new int[A.length][A[0].length];
	if (checkMat(A)){
	    for (int i = 0; i < A.length; i++) {
	      for (int j = 0; j < A[i].length; j++) {
	        result[i][j] = p * A[i][j];
	      }
	    }
	} else {
		System.out.println("The input A is not a valid matrix");
		//Throw an exception
	}

	return result;
  }

  /*Scalar Multiplication*/
  static int [][] mult (int[][] A, int p){
	if (checkMat(A)){
	    int[][] result = new int[A.length][A[0].length];
	    for (int i = 0; i < A.length; i++) {
	      for (int j = 0; j < A[i].length; j++) {
	        result[i][j] = p * A[i][j];
	      }
	    }
	    return result;
	} else {
		System.out.println("The input A is not a valid matrix");
		//Throw an exception
	}

	return null;
  }

  /*Matrix Addition*/
  static int [][] add (int[][] A, int[][] B){
	  int[][] result = new int[ A[0].length ][ B.length ];
	  if (sameSize(A,B)){
	    for (int i = 0; i < A.length; i++) {
	        for (int j = 0; j < A[i].length; j++) {
	          result[i][j] = A[i][j] + B[i][j];
	        }
	    }
	  } else {
		System.out.println("A and B cannot be added together");
		//Throw exception
	  }

	  return result;
  }

  /*Matrix Multiplication*/
  static int [][] mult (int[][] A, int[][] B){
      int[][] result = new int[ A[0].length ][ B.length ];
	  if ( checkMat(A) && checkMat(B) && canMult(A, B) ) {
    	int multSum = 0;
        for (int i = 0; i < A.length; i++) {
            for (int j = 0; j < A[i].length; j++) {
            	for (int k = 0; k < A[0].length; k++) {
                multSum += ( A[i][k] * B[k][j] );
              }
              result[i][j] = multSum;
              multSum = 0;
            }
          }
      } else {
        System.out.println("A and B cannot be multiplied");
      }
	return result;
  }

  /*Ensures that the given matrix has a valid size*/
  static boolean checkMat (int [][] A) {

	  boolean result = false;

	  for (int i = 0; i < A.length; i++) {
		  if (A[i].length == A[0].length) {
			  /*Matrix is valid*/
			  result = true;
		  } else {
			  /*Matrix is invalid*/
			  result = false;
			  break;
		  }
	  }

	  if (result) {
		  System.out.println("Matrix is valid");
	  } else {
		  System.out.println("Matrix is invalid");
	  }

	  return result;
  }

  /*Checks to see if two matrices are the same dimensions*/
  static boolean sameSize (int[][] A, int[][] B) {
	  boolean result = false;
	  if (checkMat(A) && checkMat(B)) {
		  if (A.length == B.length && A[0].length == B[0].length) {
			  result = true;
		  }
	  } else {
		  result = false;
	  }
	  return result;
  }

  /*Checks to see if two matrices are able to be multiplied*/
  static boolean canMult (int[][] A, int[][] B) {
	  boolean result = false;
	  if (checkMat(A) && checkMat(B) && A[0].length == B.length) {
		  result = true;
	  } else {
		  result = false;
	  }
	  return result;
  }

}
